post-merge epic: update pow page (#7105)
ethereum/ethereum-org-website: src/content/developers/docs/consensus-mechanisms/pow/index.md

description: An explanation of the proof-of-work consensus protocol and its role in Ethereum.
lang: en
sidebar: true
---

> Ethereum was always intended to be a [proof-of-stake](/developers/docs/consensus-mechanisms/pos) network, but the research and development required to implement proof-of-stake safely, securely and performantly took several years. The network was therefore started with a different consensus mechanism called **[Proof-of-work (PoW)](https://wikipedia.org/wiki/Proof_of_work)**. This allowed the nodes of the Ethereum network to agree on the state of all information recorded on the Ethereum blockchain and prevents certain kinds of economic attacks.

Reviewer: Do you think the intro talking about proof-of-stake first might be confusing? I feel like we should introduce proof-of-work first (Ethereum originally used proof-of-work.... proof-of-work is... it has switched to proof-of-stake).

description: An explanation of the proof-of-work consensus protocol and its role in Ethereum.
lang: en
sidebar: true
---

> The Ethereum network started using a consensus mechanism called **[Proof-of-work (PoW)](/developers/docs/consensus-mechanisms/pow)**. This allowed the nodes of the Ethereum network to agree on the state of all information recorded on the Ethereum blockchain and prevents certain kinds of economic attacks. However, Ethereum switched off proof-of-work in 2022 and started using [proof-of-stake](/developers/docs/consensus-mechanisms/pos) instead.

Reviewer 2: ```suggestion
The Ethereum network began by using a consensus mechanism that involved **[Proof-of-work (PoW)](/developers/docs/consensus-mechanisms/pow)**. This allowed the nodes of the Ethereum network to agree on the state of all information recorded on the Ethereum blockchain and prevented certain kinds of economic attacks. However, Ethereum switched off proof-of-work in 2022 and started using [proof-of-stake](/developers/docs/consensus-mechanisms/pos) instead.
```

> Ethereum was always intended to be a [proof-of-stake](/developers/docs/consensus-mechanisms/pos) network, but the research and development required to implement proof-of-stake safely, securely and performantly took several years. The network was therefore started with a different consensus mechanism called **[Proof-of-work (PoW)](https://wikipedia.org/wiki/Proof_of_work)**. This allowed the nodes of the Ethereum network to agree on the state of all information recorded on the Ethereum blockchain and prevents certain kinds of economic attacks.

Author: ```suggestion
The Ethereum network was started with a consensus mechanism called **[Proof-of-work (PoW)](https://wikipedia.org/wiki/Proof_of_work)**. This allowed the nodes of the Ethereum network to agree on the state of all information recorded on the Ethereum blockchain and prevents certain kinds of economic attacks. However, Ethereum switched off proof-of-work in 2022 and started using [proof-of-stake](/developers/docs/consensus-mechanisms/pos) instead.
```

<InfoBanner emoji=":wave:">

>     Proof-of-work has now been deprecated. Ethereum no longer uses proof-of-work as a consensus mechanism. Instead, it uses proof-of-stake. Read more on <a href="/developers/docs/consensus-mechanisms/pos/">proof-of-stake</a> and <a href="/staking/">staking</a>.    

Reviewer 2: ```suggestion
    Proof-of-work has now been deprecated. Ethereum no longer uses proof-of-work as part of its consensus mechanism. Instead, it uses proof-of-stake. Read more on <a href="/developers/docs/consensus-mechanisms/pos/">proof-of-stake</a> and <a href="/staking/">staking</a>.    
```

</InfoBanner>

## Prerequisites {#prerequisites}

To better understand this page, we recommend you first read up on [transactions](/developers/docs/transactions/), [blocks](/developers/docs/blocks/), and [consensus mechanisms](/developers/docs/consensus-mechanisms/).

## What is Proof-of-work (PoW)? {#what-is-pow}

> Proof-of-work is the mechanism that once allowed the decentralized Ethereum network to come to consensus (i.e. all nodes agree) on things like account balances and the order of transactions. This prevented users from "double spending" their coins and ensured that the Ethereum chain was tremendously difficult to attack or manipulate. These security properties now come from proof-of-stake instead.

Reviewer 2: Similar to some other comments. I think it would be a good idea to try and avoid directly calling PoS or PoW the consensus mechanism, since it is only one piece of consensus.

```suggestion
Nakamoto consensus, which utilizes proof-of-work, is the mechanism that once allowed the decentralized Ethereum network to come to consensus (i.e. all nodes agree) on things like account balances and the order of transactions. This prevented users from "double spending" their coins and ensured that the Ethereum chain was tremendously difficult to attack or manipulate. These security properties now come from proof-of-stake instead using the consensus mechanism known as [Gasper](/developers/docs/consensus-mechanisms/pos/gasper/).
```

## Proof-of-work and mining {#pow-and-mining}

> Proof-of-work is the underlying algorithm that sets the difficulty and rules for the work miners do on proof-of-work blockchains. Mining is the "work" itself. It's the act of adding valid blocks to the chain. This is important because the chain's length helps the network follow the correct fork of the blockchain. The more "work" done, the longer the chain, and the higher the block number, the more certain the network can be of the current state of things.

Reviewer 2: Similarly here, "mining" is just the act of finding the nonce that results in a block hash with an adequate number of leading binary zeros by brute-force hashing. By finding this nonce, it provide proof that the work was performed. I believe it would be more correct to say the underlying algorithm that adjusts the difficulty is the Nakamoto Consensus mechanism which uses PoW.
@jmcook1186 Thought on this?

Author: not sure - in Geth the difficulty calculation is part of the Ethash package - it's just a simple comparison between the change in time taken to mine a block relative to its parent and a threshold. To me it feels fine to describe it as part of PoW.

[More on mining](/developers/docs/consensus-mechanisms/pow/mining/)

## How did Ethereum's proof-of-work work? {#how-it-works}

> Ethereum transactions are processed into blocks. In proof-of-work Ethereum, each block contained:

Reviewer 2: Just a thought to make it more clear this is gone (in a post-merge environment)
```suggestion
Ethereum transactions are processed into blocks. In the now-deprecated proof-of-work Ethereum, each block contained:
```

- block difficulty – for example: 3,324,092,183,262,715
- mixHash – for example: `0x44bca881b07a6a09f83b130798072441705d9a665c5ac8bdf2f39a3cdf3bee29`
- nonce – for example: `0xd3ee432b4fb3d26b`

This block data was directly related to proof-of-work.

### The work in proof-of-work {#the-work}

The proof-of-work protocol, Ethash, required miners to go through an intense race of trial and error to find the nonce for a block. Only blocks with a valid nonce could be added to the chain.

> When racing to create a block, a miner repeatedly put a dataset, that could only be obtained by downloading and running the full chain (as a miner does), through a mathematical function. The dataset was used to generate a mixHash below a target nonce, as dictated by the block difficulty. The best way to do this is through trial and error.

Reviewer 2: > The dataset was used to generate a mixHash **below a target nonce**

This is a little confusing... the _nonce itself_ can be as high as it needs to be to allow the resulting block _hash_ to be less than a calculated target.

Author: yep, thanks for the catch.
```suggestion
When racing to create a block, a miner repeatedly put a dataset, that could only be obtained by downloading and running the full chain (as a miner does), through a mathematical function. The dataset was used to generate a mixHash below a target that is dictated by the block difficulty. The best way to do this is through trial and error.
```

The difficulty determined the target for the hash. The lower the target, the smaller the set of valid hashes. Once generated, this was incredibly easy for other miners and clients to verify. Even if one transaction were to change, the hash would be completely different, signalling fraud.

Hashing makes fraud easy to spot. But proof-of-work as a process was also a big deterrent to attacking the chain.

### Proof-of-work and security {#security}

> Miners were incentivised to do this work on the main Ethereum chain. There was little incentive for a subset of miners to start their own chain – it undermines the system. Blockchains rely on having a single state as a source of truth.

Reviewer 2: ```suggestion
Miners were incentivized to do this work on the main Ethereum chain. There was little incentive for a subset of miners to start their own chain—it undermines the system. Blockchains rely on having a single state as a source of truth.
```

The objective of proof-of-work was to extend the chain. The longest chain was most believable as the valid one because it had the most computational work done to generate it. Within Ethereum's PoW system, it was nearly impossible to create new blocks that erase transactions, create fake ones, or maintain a second chain. That's because a malicious miner would have needed to always solve the block nonce faster than everyone else.

To consistently create malicious yet valid blocks, a malicious miner would have needed over 51% of the network mining power to beat everyone else. That amount of "work" requires a lot of expensive computing power and the energy spent might even have outweighed the gains made in an attack.

### Proof-of-work economics {#economics}

Proof-of-work was also responsible for issuing new currency into the system and incentivizing miners to do the work.

> Miners who successfully create a block were rewarded with two freshly minted ETH and part of the transaction fees. 1.75 ETH was also paid for an uncle block. Uncle blocks are valid blocks created by a miner practically at the same time as another miner mined the successful block. Uncle blocks usually happen due to network latency.

Reviewer 2: - "uncle" -> "ommer"
- Could mention Constantinople upgrade for historical reference on the 2 ETH reward. 
- Moved "1.75" from the beginning of the sentence.
- Added note about how one becomes canonical over the other
- Switched to past-tense

```suggestion
Since the [Constantinople upgrade](/history/#constantinople), miners who successfully create a block were rewarded with two freshly minted ETH and part of the transaction fees. Ommer blocks also compensated 1.75 ETH. Ommer blocks were valid blocks created by a miner practically at the same time as another miner created the canonical block, which was ultimately determined by which chain was built on top of first. Ommer blocks usually happened due to network latency.
```

## Finality {#finality}

A transaction has "finality" on Ethereum when it's part of a block that can't change.

Because miners worked in a decentralized way, two valid blocks could be mined at the same time. This creates a temporary fork. Eventually, one of these chains became the accepted chain after subsequent blocks were mined and added to it, making it longer.

> But to complicate things further, transactions rejected on the temporary fork may have been included in the accepted chain. This means it could get reversed. So finality refers to the time you should wait before considering a transaction irreversible. For Ethereum, the recommended time was six blocks or just over 1 minute. After six blocks, there was high confidence that the transaction was successful - lonmgher waits gave higher confidence. Now, with proof-of-stake, finalization is an explicit, rather than probabilistic, property of a block.

Reviewer 2: I think you're trying to suggest that the transaction may _not_ be included in the accepted chain:
```suggestion
To complicate things further, transactions rejected on the temporary fork may not have been included in the accepted chain. This means it could get reversed. So finality refers to the time you should wait before considering a transaction irreversible. Under the previous proof-of-work Ethereum, the recommended time was six blocks or just over 1 minute. After six blocks, there was high confidence that the transaction was successful—longer waits gave higher confidence. Now, with proof-of-stake, finalization is an explicit, rather than probabilistic, property of a block.
```

Reviewer 2: > the recommended time was six blocks or just over 1 minute

Also, where did you get this recommendation from? Many exchanges, for example, wait closer to 30 blocks to consider a tx finalized. 

In either case, I wonder if it's worth expanding the last sentence and mentioning that under PoS "finality" will not necessary be any faster in a practical sense, but it does allow an official "finalized" state, unlike PoW which just gets less and less probable with each successive block.

Author: it was in the original page - it's not my addition.

Author: ```suggestion
To complicate things further, transactions rejected on the temporary fork may not have been included in the accepted chain. This means it could get reversed. So finality refers to the time you should wait before considering a transaction irreversible. Under the previous proof-of-work Ethereum, the more blocks were mined on top of a specific block `N`, the higher confidence that the transactions in `N` were successful and would not be reverted. Now, with proof-of-stake, finalization is an explicit, rather than probabilistic, property of a block.
```

## Proof-of-work energy-usage {#energy}

> A major criticism of proof-of-work is the amount of energy output required to keep the network safe. To maintain security and decentralization, Ethereum on proof-of-work consumed large amounts of energy. Shortly before switching to proof-of-stake, Ethereum miners were collectively consuming about 70 TWh/yr (about the same as the Czech Republic - according to [digiconomist](digiconomist.net) on 18/07/22).

Reviewer 2: For sake of internationalization, I personally try to state dates with the month spelled out to avoid confusion between MM/DD/YY and DD/MM/YY

Would consider:
```suggestion
A major criticism of proof-of-work is the amount of energy output required to keep the network safe. To maintain security and decentralization, Ethereum on proof-of-work consumed large amounts of energy. Shortly before switching to proof-of-stake, Ethereum miners were collectively consuming about 70 TWh/yr (about the same as the Czech Republic - according to [digiconomist](digiconomist.net) on 18-July-2022).
```

(I know there is no "month 18" in this case, just may be a good habit for us to get into)
